.github/workflows/release.yml

```
@@ -30,15 +35,26 @@ jobs:
+                  # RELEASE_FILES is the single source of truth for the release
+                  read -r -a release_files <<< "$RELEASE_FILES"
+                  for f in "${release_files[@]}"; do
+                      test -s "$f" || { echo "error: $f missing or empty"; exit 1; }
+                  done
+                  echo "all release assets present"
+
             - name: Attest build artifacts
               uses: actions/attest-build-provenance@e8998f949152b193b063cb0ec769d69d929409be # v4.1.1
               with:
-                  subject-path: "main.js,manifest.json"
+                  subject-path: ${{ replace(env.RELEASE_FILES, ' ', ',') }}
```

actionlint (1.7.12)

[error] undefined function "replace". available functions are "always", "cancelled", "case", "contains", "endswith", "failure", "format", "fromjson", "hashfiles", "join", "startswith", "success", "tojson"

(expression)